api/pkg/db/initializer/initializer.go

```diff
@@ -82,13 +82,29 @@ type initFn func(*gorm.DB, *log.Logger, *app.Data) error
 
 func addCategories(db *gorm.DB, log *log.Logger, data *app.Data) error {
 
+	var configCatID []uint
 	for _, c := range data.Categories {
 		cat := model.Category{Name: c.Name}
 		if err := db.Where(cat).FirstOrCreate(&cat).Error; err != nil {
 			log.Error(err)
 			return err
 		}
+		configCatID = append(configCatID, cat.ID)
 	}
+
+	if len(configCatID) > 0 {
+		// Deletes mapping of removed categories and resources from the database
+		if err := db.Unscoped().Not(map[string]interface{}{"category_id": configCatID}).Delete(&model.ResourceCategory{}).Error; err != nil {
+			log.Error(err)
+			return err
+		}
+		// Deletes categories from database which are removed from the config
+		if err := db.Unscoped().Not(map[string]interface{}{"id": configCatID}).Delete(&model.Category{}).Error; err != nil {
```

Comment:
we need to handle the case if the categories is still mapped to any resource

Reply:
so in that case categories and their mapping with resource shouldn't delete, right ?

Reply:
well in that case the the change shouldn't have made in config.yaml
because we are commiting the change in repo and then saying not to delete it from db

Reply:
for hub source of truth is config.yaml .. if the category is removed/renamed from config that means we don't want to show that on hub so mapping can be deleted..
when the category is change in catalog.. new mapping with new categories will get added in db

Reply:
hmm, cool. So those esources will turn in other categories then right